testing/tests/DevExpress.exporter/jspdfParts/jspdf.dataGrid.tests.js

```diff
@@ -30,51 +29,76 @@ const moduleConfig = {
+        exportDataGrid(getOptions(this, dataGrid)).then((jsPDFDocument) => {
             const autoTableOptions = jsPDFDocument.autoTable.__autoTableOptions;
-            helper.checkTableAndColumnWidths(undefined, [], autoTableOptions);
             helper.checkCellsContent([], [], autoTableOptions);
             done();
         });
     });
 
     QUnit.test('Simple grid', function(assert) {
         const done = assert.async();
-        const ds = [{ id: 1, name: 'test' }];
+        const ds = [{ prop1: 'text1' }];
         const dataGrid = $('#dataGrid').dxDataGrid({
             dataSource: ds,
-            keyExpr: 'id',
-            columns: [
-                { dataField: 'id', width: 50, caption: 'id' },
-                { dataField: 'name', caption: 'name' }
-            ],
+            columns: [{ dataField: 'prop1', caption: 'f1' }],
             loadingTimeout: undefined,
-            showColumnHeaders: true,
-            width: 500
+            showColumnHeaders: true
         }).dxDataGrid('instance');
-        const expectedCells = [];
-        exportDataGrid(getOptions(this, dataGrid, expectedCells)).then((jsPDFDocument) => {
+
+        exportDataGrid(getOptions(this, dataGrid)).then((jsPDFDocument) => {
             const autoTableOptions = jsPDFDocument.autoTable.__autoTableOptions;
-            helper.checkTableAndColumnWidths(Export.convertPixelsToPoints(500), [Export.convertPixelsToPoints(50), 'auto'], autoTableOptions);
-            helper.checkCellsContent([['id', 'name']], [['1', 'test']], autoTableOptions);
+            helper.checkCellsContent([['f1']], [['text1']], autoTableOptions);
             done();
         });
     });
+
+    [true, false].forEach((keepColumnWidths) => {
+        QUnit.test(`Table and column widths, keepColumnWidths: ${keepColumnWidths}`, function(assert) {
+            const done = assert.async();
+            const ds = [{ id: 1, name: 'test' }];
+            const dataGrid = $('#dataGrid').dxDataGrid({
+                dataSource: ds,
+                keyExpr: 'id',
+                columns: [
+                    { dataField: 'id', width: 50, caption: 'id' },
+                    { dataField: 'name', caption: 'name' }
+                ],
+                loadingTimeout: undefined,
+                showColumnHeaders: true,
+                width: 500
+            }).dxDataGrid('instance');
+            const options = {
+                autoTableOptions: {
+                    tableWidth: 500
+                },
+                keepColumnWidths: keepColumnWidths
+            };
+
+            exportDataGrid(getOptions(this, dataGrid, options)).then((jsPDFDocument) => {
+                const autoTableOptions = jsPDFDocument.autoTable.__autoTableOptions;
+                const expectedColumnWidths = keepColumnWidths ? [50, 450] : ['auto', 'auto'];
+                helper.checkColumnWidths(expectedColumnWidths, autoTableOptions);
```

Comment:
lets add actualAutoTableOptions.tableWidth to test

Reply:
ok